content/blog/case-weights/index.Rmd

```diff
@@ -0,0 +1,310 @@
+
+Frequency weights are more complex. Suppose we are using 10-fold cross-validation with the logistic regression on the UCB admission data, should we:
+
+* Have all the case weights be placed into either the modeling or holdout set? 
+* Fractionally, split the case weights into both the modeling and holdout data? 
+
+For the latter case, suppose a row of data has a case weight of 100 and we use 10-fold cross-validation. We would always put 90 of those 100 into the modeling data set and the other 10 into the holdout. This seems to be consistent with the sampling of the data and is what would happen if there were actually 100 rows in the data (instead of one row with a case weight of 100). However, it does raise questions regarding data leakage by just re-predicting the same data that went into the model. 
+
+This is also an area where we'd like [community feedback](https://community.rstudio.com/). 
+
+## Tidymodels syntax
+
+Let's work through an example. We'll use some data simulated with a severe class imbalance. These functions are in the new version of the modeldata package.
+
+```{r sim-data}
+set.seed(1)
+training_sim <- sim_classification(5000, intercept = -25) 
+training_sim %>% count(class)
+```
+
+If we would like to encourage models to more accurately predict the minority class, we can give these samples a much larger weight in the analysis
+
+```{r sim-wts}
+training_sim <-
+  training_sim %>% 
+  mutate(
+    case_wts = ifelse(class == "class_1", 60, 1),
+    case_wts = importance_weights(case_wts)
+  )
+```
+
+We strongly advise that users set the case weight column before any other tidymodels functions are used. This ensures that they are handled correctly in the analyses that follow. 
+
+Let's use 10-fold cross-validation to resample the data. This case is unaffected by the presence of weights: 
+
+```{r sim-folds}
+set.seed(2)
+sim_folds <- vfold_cv(training_sim, strata = class)
+```
+
+We'll fit a regularized logistic regression model to the data using glmnet: 
+
+```{r sim-model}
+lr_spec <- 
+  logistic_reg(penalty = tune(), mixture = 1) %>% 
+  set_engine("glmnet")
+```
+
+For this model, we need to ensure that the predictors are in the same units. We'll use a recipe to center and scale the data and also add some spline terms for predictors that appear to have a nonlinear relationship with the outcome: 
+
+```{r sim-rec}
+sim_rec <- 
+  recipe(class ~ ., data = training_sim) %>% 
+  step_ns(starts_with("non_linear"), deg_free = 10) %>% 
+  step_normalize(all_numeric_predictors())
+sim_rec
+```
+
+There are a few things to point out here. The recipe automatically detects the case weights even though they are captured by the dot in the right-hand side of the formula. The recipe automatically sets their role and will error if that column is changed in any way. 
+
+As mentioned above, any unsupervised steps are unaffected by importance weights so neither `step_ns()` or `step_normalize()` use the weights in their calculations. 
```

Comment:
`step_ns()` doesn't have case weights support at all, and `step_normalize()` does use importance weights in the current implementation.

Reply:
This is an error then. The code checks to see if the step is supervised or not and unsupervised steps are not supposed to use the weights.